src/esp/gfx/RenderTarget.cpp

```diff
@@ -83,34 +88,37 @@ struct RenderTarget::Impl {
     }
 
     framebuffer_ = Mn::GL::Framebuffer{{{}, size}};
-    if (flags_ & Flag::RgbaBuffer) {
-      framebuffer_.attachRenderbuffer(RgbaBuffer, colorBuffer_);
+    if (flags_ & Flag::RgbaAttachment) {
+      framebuffer_.attachRenderbuffer(RgbaBufferAttachment, colorBuffer_);
     }
-    if (flags_ & Flag::ObjectIdBuffer) {
-      framebuffer_.attachRenderbuffer(ObjectIdBuffer, objectIdBuffer_);
+    if (flags_ & Flag::ObjectIdAttachment) {
+      framebuffer_.attachTexture(ObjectIdTextureColorAttachment,
+                                 objectIdTexture_, 0);
     }
-    if (flags_ & Flag::DepthTexture) {
+    if (flags_ & Flag::DepthTextureAttachment) {
       framebuffer_.attachTexture(Mn::GL::Framebuffer::BufferAttachment::Depth,
                                  depthRenderTexture_, 0);
     } else {
       framebuffer_.attachRenderbuffer(
           Mn::GL::Framebuffer::BufferAttachment::Depth, unprojectedDepth_);
     }
-    if (flags_ & Flag::RgbaBuffer) {
-      framebuffer_.mapForDraw({{0, RgbaBuffer}});
+    if (flags_ & Flag::RgbaAttachment) {
+      framebuffer_.mapForDraw(
+          {{Mn::Shaders::Generic3D::ColorOutput, RgbaBufferAttachment}});
     }
-
-    if (flags_ & Flag::ObjectIdBuffer) {
-      framebuffer_.mapForDraw({{1, ObjectIdBuffer}});
+    if (flags_ & Flag::ObjectIdAttachment) {
```

Comment:
@mosra:
There is a big bug hidden here that even you did not capture it.
When I need to visualize the semantic info, both `flags_ & Flag::RgbaAttachment` and `flags_ & Flag::ObjectIdAttachment` will be true.
That means `mapForDraw` will be called twice in two different if clauses. So the first attempt is overidden by the second. Sigh...